### PR TITLE
E2E fixes for Gutenberg 10.6.x / 10.7.x

### PR DESCRIPTION
**Pull request by fullofcaffeine**
Tracking issue: https://github.com/Automattic/wp-calypso/issues/52595, https://github.com/Automattic/wp-calypso/issues/53135.

Trigger the tests in TeamCity from this branch.

### CODE REVIEW

**Comment by WunderBart** [comment]
Looks like there's only [1 test failing](https://teamcity.a8c.com/viewLog.html?buildId=6195314&buildTypeId=calypso_WPComTests_gutenberg_desktop) - the Premium Block is invalidating. This might be fixed via https://github.com/Automattic/jetpack/pull/19844 so let's wait until it's in. cc @fullofcaffeine @apeatling

**Review by scinos — APPROVED** [review]

**Comment by fullofcaffeine** [comment]
This can't be merged until >= 10.6 is on production, as it includes a test fix that's specific to that version (the `__experimental` to `__unstable` API change) and supporting both in the test wouldn't be worth the effort, I think.

File: test/e2e/lib/gutenberg/gutenberg-editor-component.js [hunk 1]
```diff
@@ -387,8 +385,10 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 			title
 		);
 
+		// @todo Remove the deprecated query once GB v10.7+ is up on prod. This is needed for now to make sure all tests (more specifically upgrade tests that use non-edge sites) pass.
+		const insertedBlockItemDeprecatedQuery = `.edit-post-layout__inserter-panel .block-editor-block-types-list button.editor-block-list-item-${ prefix }${ blockClass }`;
```
Comment (fullofcaffeine) [inline test/e2e/lib/gutenberg/gutenberg-editor-component.js:389]: Ops, this should have been named "inserter" not "inserted" 🤦🏻. It didn't affect the actual fix, but wasn't the correct name. Good thing is that we'll remove it now that v10.7.1 is on WPCOM.

Comment (fullofcaffeine) [reply test/e2e/lib/gutenberg/gutenberg-editor-component.js:389]: Removed in https://github.com/Automattic/wp-calypso/pull/53395.

**Pull request merged**